src/api/v1/project/project_controllers.js

```diff
@@ -45,15 +47,78 @@ export const GetProjectsController = (req, res, next) => {
     })
 }
 
-export async function GetProjectController(req, res, next) {
-    try {
-        const { projectId } = req.params
-        const project = await Project.findById(projectId)
+export const GetProjectController = (req, res, next) => {
+    const { projectId } = req.params
+    const query = { _id: new mongodb.ObjectId(projectId), isDeleted: false }
+    Project.findOne(query, function (err, project) {
+        if (err) {
+            var err = new Error('error occurred')
+            return next(err)
+        }
         if (!project) {
-            return next(new CustomError(404, "Project doesn't exist"))
+            return res.status(404).json({
+                message: 'Project doest not exits',
+            })
         }
-        return responseHandler(res, 200, project, 'Project retrieved successfully')
-    } catch (error) {
-        next(new InternalServerError(error))
+        return res.status(200).json({
+            status: 'success',
+            message: 'Project retrieved successfully',
+            project,
+        })
+    }).catch(error => {
+        return res.status(500).json({
+            status: 'error',
+            message: 'Something went wrong. Try again',
+            error,
+        })
+    })
+}
+
+export const UpdateProjectController = (req, res) => {
+    if (!req.body) {
+        return res.status(400).send({
+            message: 'Data to update can not be empty!',
+        })
     }
+    const { projectId } = req.params
+    const query = { _id: new mongodb.ObjectId(projectId), isDeleted: false }
+
+    Project.findOneAndUpdate(query, req.body, { useFindAndModify: false })
+        .then(project => {
+            if (!project) {
+                res.status(404).json({ message: 'Project does not exist' })
+            } else {
+                res.status(201).json({
+                    status: 'success',
+                    message: 'Project updated successfully.',
+                })
+            }
+        })
+        .catch(err => {
+            res.status(500).json({
+                message: 'Error when updating project with id=' + projectId,
+            })
+        })
+}
+
+export const DeleteProjectController = (req, res) => {
+    const { projectId } = req.params
+    const query = { _id: new mongodb.ObjectId(projectId), isDeleted: false }
+
+    Project.findOneAndUpdate(query, { isDeleted: true })
```

Comment:
grant showman access after deleting, `showmanAccess: true`
```suggestion
    Project.findOneAndUpdate(query, { isDeleted: true, showmanAccess: true })
```